SetupDataPkg/Plugins/GenSetupDataBin/GenSetupDataBin.py

```diff
@@ -123,41 +96,98 @@ def do_pre_build(self, thebuilder):
+            return -1
+
+        if delta_conf is not None and csv_conf is not None:
+            delta_conf = delta_conf.split(";")
+            csv_conf = csv_conf.split(";")
+            # Validate we have the same number of delta and csv files, as required
+            if len(delta_conf) != len(csv_conf):
+                logging.error("Differing number of Delta and CSV files provided for profiles, they must be the same.")
+                return -1
+
+            for idx in range(len(delta_conf)):
+                # Validate the name of the delta file and csv file match, to mitigate human error in placement
+                # ignore file suffix
+                if delta_conf[idx][:-4] != csv_conf[idx][:-4]:
+                    logging.error("Delta and CSV files do not have the same name, list might be in the wrong order.")
+                    return -1
+
+                # Generate the profile
+                self.generate_profile(thebuilder, delta_conf[idx], csv_conf[idx], idx + 1)
+        elif delta_conf is not None:
+            delta_conf = delta_conf.split(";")
+
+            for idx in range(len(delta_conf)):
+                # Generate the profile
+                self.generate_profile(thebuilder, delta_conf[idx], None, idx + 1)
+        elif csv_conf is not None:
+            csv_conf = csv_conf.split(";")
+
+            for idx in range(len(csv_conf)):
+                # Generate the profile
+                self.generate_profile(thebuilder, None, csv_conf[idx], idx + 1)
+        else:
+            logging.warn("Either DELTA_CONF_POLICY or CSV_CONF_POLICY or both not set. Only generic profile generated.")
+
+        return 0
+
+    # Delete intermediate binary files
     #
-    # Consumes build environment variables: "MSFT_PLATFORM_PACKAGE"
+    # Consumes build environment variables: "DELTA_CONF_POLICY" and "CSV_CONF_POLICY"
     def do_post_build(self, thebuilder):
-        if thebuilder.env.GetValue("MSFT_PLATFORM_PACKAGE") is not None:
-            os.remove(thebuilder.mws.join(thebuilder.ws, thebuilder.env.GetValue("MSFT_PLATFORM_PACKAGE"),
-                                          "BuiltInVars.xml"))
+        op_dir = thebuilder.mws.join(thebuilder.ws, thebuilder.env.GetValue("BUILD_OUTPUT_BASE"), "ConfPolicy")
+        yaml_conf_file = thebuilder.env.GetValue("YAML_CONF_FILE")
+        xml_conf_file = thebuilder.env.GetValue("XML_CONF_FILE")
+        delta_conf = thebuilder.env.GetValue("DELTA_CONF_POLICY")
+        csv_conf = thebuilder.env.GetValue("CSV_CONF_POLICY")
+
+        if yaml_conf_file is not None:
+            yaml_filename = os.path.join(op_dir, "YAMLPolicyVarBin_0.bin")
+            os.remove(yaml_filename)
+
+        if xml_conf_file is not None:
+            xml_filename = os.path.join(op_dir, "XMLPolicyVarBin_0.bin")
+            os.remove(xml_filename)
+
+        if delta_conf is not None and csv_conf is not None:

```

Comment:
this is fine as is, but would it be easier to remove the "elif" and just check `delta_conf`, then remove `yaml_filename`, then check `csv_conf` and remove `xml_filename`?

Reply:
Agreed, copy/paste from above. I'll simplify.